components/payments/internal/connectors/engine/engine.go

```diff
@@ -229,6 +278,147 @@ func (e *engine) HandleWebhook(ctx context.Context, urlPath string, webhook mode
 	return nil
 }
 
+func (e *engine) CreatePool(ctx context.Context, pool models.Pool) error {
+	if err := e.storage.PoolsUpsert(ctx, pool); err != nil {
+		return err
+	}
+
+	ctx, _ = contextutil.Detached(ctx)
```

Comment:
This is only creating one workflow. Does it need to be detached?

Reply:
Since we already inserted the pool inside the database, I wanted to be sure that the event is sent also, hence the detached context. What do you think ?